rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -583,14 +628,15 @@ u64 debugger_frame::EvaluateExpression(const QString& expression)
 {
 	auto thread = cpu.lock();
 
-	if (!thread) return 0;
-
 	bool ok = false;
 
+	const u32 pc = get_rsx() ? +m_rsx->ctrl->get
+		: (thread ? thread->get_pc() : 0);
+
 	// Parse expression(or at least used to, was nuked to remove the need for QtJsEngine)
 	const QString fixed_expression = QRegExp("^[A-Fa-f0-9]+$").exactMatch(expression) ? "0x" + expression : expression;
 	const u64 res = static_cast<u64>(fixed_expression.toULong(&ok, 16));
-	return ok ? res : thread->get_pc();
+	return ok ? res : pc;
```

Comment:
Why not write something like:
if (ok) return res;
if (auto thread = get_rsx()) return +thread->ctrl->get;
auto thread = cpu.lock();
if (thread) return thread->get_pc();
return 0;

Reply:
Fixed.